packages/@aws-cdk/aws-ec2/lib/security-group.ts

```diff
@@ -294,6 +296,31 @@ export interface SecurityGroupImportOptions {
  * ```
  */
 export class SecurityGroup extends SecurityGroupBase {
+  /**
+   * Look up a security group by id.
+   */
+  static fromLookup(scope: Construct, id: string, securityGroupId: string) {
+    const attributes: cxapi.SecurityGroupContextResponse = ContextProvider.getValue(scope, {
+      provider: cxschema.ContextProvider.SECURITY_GROUP_PROVIDER,
+      props: {
+        securityGroupId: securityGroupId,
+      } as cxschema.SecurityGroupContextQuery,
+      dummyValue: undefined,
+    }).value;
+
+    // Dummy attributes for tests and the first pass.
+    const dummyAttributes: cxapi.SecurityGroupContextResponse = {
+      securityGroupId: securityGroupId,
+      allowAllOutbound: true,
+    };
+
+    const attributesToUse = attributes ?? dummyAttributes;
```

Comment:
Only because I found something similar in `Vpc.fromLookup()`, where the dummy value is held separately and not given as `dummyValue` input to `getValue`. Given that I don't have strong feelings here, I'll change this to just use `dummyValue` flat out.

Reply:
The VPC is the odd one out. There was a specific subnet-related problem there that needed to be special-cased (see #4090) .